java/algorithm/src/test/java/org/apache/arrow/algorithm/sort/TestSortingUtil.java

```diff
@@ -0,0 +1,150 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.sort;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+import java.util.Random;
+import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+
+import org.apache.arrow.vector.BigIntVector;
+import org.apache.arrow.vector.Float4Vector;
+import org.apache.arrow.vector.Float8Vector;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.SmallIntVector;
+import org.apache.arrow.vector.TinyIntVector;
+import org.apache.arrow.vector.ValueVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.testing.ValueVectorDataPopulator;
+
+/**
+ * Utilities for sorting related utilities.
+ */
+public class TestSortingUtil {
+
+  static final Random random = new Random(0);
+
+  static final DataGenerator<TinyIntVector, Byte> TINY_INT_GENERATOR = new DataGenerator<>(
+      () -> (byte) random.nextInt(),
+      (vector, array) -> ValueVectorDataPopulator.setVector(vector, array), Byte.class);
+
+  static final DataGenerator<SmallIntVector, Short> SMALL_INT_GENERATOR = new DataGenerator<>(
+      () -> (short) random.nextInt(),
+      (vector, array) -> ValueVectorDataPopulator.setVector(vector, array), Short.class);
+
+  static final DataGenerator<IntVector, Integer> INT_GENERATOR = new DataGenerator<>(
+      () -> random.nextInt(),
+      (vector, array) -> ValueVectorDataPopulator.setVector(vector, array), Integer.class);
+
+  static final DataGenerator<BigIntVector, Long> LONG_GENERATOR = new DataGenerator<>(
+      () -> random.nextLong(),
+      (vector, array) -> ValueVectorDataPopulator.setVector(vector, array), Long.class);
+
+  static final DataGenerator<Float4Vector, Float> FLOAT_GENERATOR = new DataGenerator<>(
+      () -> random.nextFloat(),
+      (vector, array) -> ValueVectorDataPopulator.setVector(vector, array), Float.class);
+
+  static final DataGenerator<Float8Vector, Double> DOUBLE_GENERATOR = new DataGenerator<>(
+      () -> random.nextDouble(),
+      (vector, array) -> ValueVectorDataPopulator.setVector(vector, array), Double.class);
+
+  static final DataGenerator<VarCharVector, String> STRING_GENERATOR = new DataGenerator<>(
+      () -> {
+        int strLength = random.nextInt(20) + 1;
+        byte[] str = new byte[strLength];
+        random.nextBytes(str);
+        return new String(str);
```

Comment:
I think that this can generate strings that are not valid UTF-8. `VarCharVector` does not enforce valid UTF-*, but the Arrow specification does specify UTF-8, so it could be confusing for someone who reads this code.

Reply:
Sounds reasonable. I have revised the code so that random strings can only contain alphabetic characters.